pkg/render/goldmane/component.go

```diff
@@ -0,0 +1,280 @@
+	GoldmaneServiceName   = "goldmane"
+)
+
+func Goldmane(cfg *Configuration) render.Component {
+	c := &Component{cfg: cfg}
+
+	return c
+}
+
+// Configuration contains all the config information needed to render the component.
+type Configuration struct {
+	PullSecrets                 []*corev1.Secret
+	OpenShift                   bool
+	Installation                *operatorv1.InstallationSpec
+	TrustedCertBundle           certificatemanagement.TrustedBundleRO
+	GoldmaneServerKeyPair       certificatemanagement.KeyPairInterface
+	ManagementClusterConnection *operatorv1.ManagementClusterConnection
+	ClusterDomain               string
+}
+
+type Component struct {
+	cfg *Configuration
+
+	goldmaneImage string
+}
+
+func (c *Component) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := c.cfg.Installation.Registry
+	path := c.cfg.Installation.ImagePath
+	prefix := c.cfg.Installation.ImagePrefix
+
+	var err error
+
+	c.goldmaneImage, err = components.GetReference(components.ComponentCalicoGoldmane, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Component) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *Component) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		c.serviceAccount(),
+		c.role(),
+		c.roleBinding(),
+		c.goldmaneService(),
+		c.deployment(),
+		c.networkPolicy(),
+	}
+
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(GoldmaneNamespace, c.cfg.PullSecrets...)...)...)
+
+	// Goldmane needs to be removed if the installation is not Calico, since it's not supported (yet!) for any other variant.
+	if c.cfg.Installation.Variant == operatorv1.Calico {
+		return objs, nil
+	} else {
+		return nil, objs
+	}
+}
+
+func (c *Component) Ready() bool {
+	return true
+}
+
+func (c *Component) serviceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta:   metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: GoldmaneServiceAccountName, Namespace: GoldmaneNamespace},
+	}
+}
+
+func (c *Component) goldmaneContainer() corev1.Container {
+	env := []corev1.EnvVar{
+		{Name: "LOG_LEVEL", Value: "INFO"},
+		{Name: "PORT", Value: fmt.Sprintf("%d", GoldmaneServicePort)},
+		{Name: "SERVER_CERT_PATH", Value: c.cfg.GoldmaneServerKeyPair.VolumeMountCertificateFilePath()},
+		{Name: "SERVER_KEY_PATH", Value: c.cfg.GoldmaneServerKeyPair.VolumeMountKeyFilePath()},
+		{Name: "CA_CERT_PATH", Value: c.cfg.TrustedCertBundle.MountPath()},
+	}
+	guardianSvc := render.GuardianService(c.cfg.ClusterDomain)
+
+	volumeMounts := []corev1.VolumeMount{c.cfg.GoldmaneServerKeyPair.VolumeMount(c.SupportedOSType())}
+	volumeMounts = append(volumeMounts, c.cfg.TrustedCertBundle.VolumeMounts(c.SupportedOSType())...)
+
+	if c.cfg.ManagementClusterConnection != nil {
+		env = append(env,
+			corev1.EnvVar{Name: "PUSH_URL", Value: fmt.Sprintf("%s/api/v1/flows/bulk", guardianSvc)},
```

Comment:
Maybe not for this PR, but we need to figure out how we're going to dynamically learn this information.

Should we always set this env var and have Goldmane determine when to push based on some other indicator?  Should we use a ConfigMap to mount in configuration and use a file watcher?